validator/sawtooth_validator/execution/scheduler_parallel.py

```diff
@@ -89,6 +98,44 @@ def _get(self, address, create=False):
     def get(self, address):
         return self._get(address)
 
+    def walk(self, address):
+        '''
+        Returns a stream of node data, raising AddressNotInTree if
+        ADDRESS is not in the tree.
+
+        First, the ancestors of ADDRESS (including itself) are
+        yielded, earliest to latest. Next, a dummy node is yielded to
+        indicate that all the ancestors have been returned. Finally,
+        the descendants of ADDRESS are yielded in an unspecified
+        order.
+        '''
+
+        node = self._root
+
+        yield node
+
+        for token in self._tokenize_address(address):
+            try:
+                node = node.children[token]
+            except KeyError:
+                raise AddressNotInTree()
```

Comment:
I think it is probably okay to just raise KeyError here if you call out in the docstring that it will raise that on missing address.